tests/test_queries.py

```diff
@@ -0,0 +1,890 @@
+            logger.info("Generate raw stats")
+            QueryTestBase.stats = generate_raw_stats(self._registry_path)
+        return QueryTestBase.stats
+
+    @abc.abstractmethod
+    def make_query(self):
+        """Return the query model"""
+
+    @abc.abstractmethod
+    def validate(self):
+        """Validate the results
+
+        Returns
+        -------
+        bool
+            Return True when the validation is successful.
+
+        """
+
+    def get_filtered_county_id(self):
+        filters = self._model.project.dimension_filters
+        counties = [x.value for x in filters if x.query_name == "county"]
+        assert len(counties) == 1, f"Unexpected length of filtered counties: {len(counties)}"
+        return counties[0]
+
+    def get_expected_ca_max_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return max(
+            (
+                by_county["06037"]["comstock_resstock"]["max"]["electricity"],
+                by_county["06073"]["comstock_resstock"]["max"]["electricity"],
+            )
+        )
+
+    def get_expected_ny_max_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return max(
+            (
+                by_county["36047"]["comstock_resstock"]["max"]["electricity"],
+                by_county["36081"]["comstock_resstock"]["max"]["electricity"],
+            )
+        )
+
+    def get_expected_ca_sum_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return (
+            by_county["06037"]["comstock_resstock"]["sum"]["electricity"]
+            + by_county["06073"]["comstock_resstock"]["sum"]["electricity"]
+        )
+
+    def get_expected_ny_sum_electricity(self):
+        by_county = self.get_raw_stats()["by_county"]
+        return (
+            by_county["36047"]["comstock_resstock"]["sum"]["electricity"]
+            + by_county["36081"]["comstock_resstock"]["sum"]["electricity"]
+        )
+
+
+class QueryTestElectricityAgg(QueryTestBase):
+
+    NAME = "electricity-agg-06037"
+
+    def __init__(self, operation, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self._operation = operation
+
+    def make_query(self):
+        name = self.NAME
+        self._model = ProjectQueryResultModel(
+            name=name,
+            project=ProjectConstraintsModel(
+                project_id="dsgrid_conus_2022",
+                include_dsgrid_dataset_components=True,
+                dataset_ids=[
+                    "conus_2022_reference_comstock",
+                    "conus_2022_reference_resstock",
+                    # "tempo_conus_2022",
+                ],
+                # TODO: need to allow min/max/sum of metric on other supplemental dimensions
+                dimension_filters=[
+                    DimensionFilterExpressionModel(
+                        dimension_type=DimensionType.GEOGRAPHY,
+                        query_name="county",
```

Comment:
Is this also a `dimension_query_name`? What is the difference between `query_name` here and `dimension_query_name` in the `MetricReductionModel`?

Reply:
They are different models that have different fields. Per our meeting I added the `dimension_query_name` alias to `MetricReductionModel` but didn't think to add it to all dimension filter models. I can do that.